specifyweb/frontend/js_src/lib/components/QueryBuilder/Fields.tsx

```diff
@@ -61,9 +62,71 @@ export function QueryFields({
     | ((line: number, direction: 'down' | 'up') => void)
     | undefined;
   readonly onOpenMap: ((line: number) => void) | undefined;
+  readonly onChangeFields?: ((fields: RA<QueryField>) => void) | undefined;
 }): JSX.Element {
   const fieldsContainerRef = React.useRef<HTMLUListElement | null>(null);
 
+  const [draggedItem, setDraggedItem] = React.useState<QueryField | null>(null);
+
+  const [isDragging, setIsDragging] = React.useState(false);
+
+  React.useEffect(() => {
```

Comment:
This is quite buggy:

https://user-images.githubusercontent.com/40512816/223826943-48baaf96-07db-4d91-b00b-08673c1f4684.mp4

Either make it less buggy, or look into using libraries that handle this very well (after doing a bit of research, found: react-dnd, https://dev.to/augani/how-to-create-a-smooth-draggable-list-with-react-in-10-mins-2fbn, https://codesandbox.io/examples/package/react-draggable-list, https://github.com/atlassian/react-beautiful-dnd, https://shopify.github.io/draggable/